sql: collated strings don't get deleted from index

If I make this seemingly innocuous change to `collatedstring_index2` test file:
```diff
@@ -154,7 +154,7 @@ statement ok
 DELETE FROM t WHERE a > ('a' COLLATE de) AND a < ('c' COLLATE de);
 
 query TI
-SELECT a, b FROM t ORDER BY (a, b);
+SELECT a, b FROM t ORDER BY a, b;
 ```
The test fails with:
```
        --- FAIL: TestLogic/default/collatedstring_index2 (0.31s)
        	logic_test.go:1589: 
        		testdata/logic_test/collatedstring_index2:156: 
        		expected:
        		    üü  5
        		    üü  6
        		    xx  5
        		    
        		but found (query options: "") :
        		    aa  2  
        		    aa  3  
        		    AA  1  
        		    bb  4  
        		    üü  5  
        		    üü  6  
        		    xx  5  
```

The difference between these two statements is that in the first one `(a, b)` is an expression, which means that we will render an `(a, b)` tuple for each row, and use that as a key to sort. This means that we can't really use the index on `a, b`. So these tests are not actually testing the index at all!! Note: in these cases, we need to include `EXPLAIN` testcases to make sure the expected indexes are used. Indeed, see the difference:

```
query ITTT
EXPLAIN SELECT a, b FROM t ORDER BY a, b;
----
0  render
1  scan
1          table  t@t_a_b_idx
1          spans  ALL

query ITTT
EXPLAIN SELECT a, b FROM t ORDER BY (a, b);
----
0  sort
0          order  +"(a, b)"
1  render
2  scan
2          table  t@primary
2          spans  ALL
```

So it looks like DELETE is not actually deleting from the index. Indeed:

```
SELECT * FROM t:
üü  5  NULL
xx  5  false
üü  6  true

SELECT * FROM t@t_a_b_idx:
aa  2  false
aa  3  true
AA  1  true
bb  4  false
üü  5  NULL
üü  6  true
xx  5  false
```

I believe this same mistake is in more of the collated string tests, so they should all be audited and EXPLAIN cases should be added where indexes are expected to be used.
CC @knz

--- a/pkg/sql/testdata/logic_test/collatedstring_index1
+++ b/pkg/sql/testdata/logic_test/collatedstring_index1
@@ -69,9 +69,7 @@ SELECT COUNT (a) FROM t WHERE a > ('a' COLLATE da) AND a < ('c' COLLATE da)
 # Create an index and try again.
 
 statement ok
-CREATE INDEX ON t (b, a)
-
-# TODO(#14533): add STORING (c)
+CREATE INDEX ON t (b, a) STORING (c)
 
 query ITTT
 EXPLAIN SELECT a, b FROM t ORDER BY a, b

--- a/pkg/sql/testdata/logic_test/collatedstring_index2
+++ b/pkg/sql/testdata/logic_test/collatedstring_index2
@@ -69,9 +69,7 @@ SELECT COUNT (a) FROM t WHERE a > ('a' COLLATE de) AND a < ('c' COLLATE de)
 # Create an index and try again.
 
 statement ok
-CREATE INDEX ON t (a, b)
-
-# TODO(#14533): add STORING (c)
+CREATE INDEX ON t (a, b) STORING (c)
 
 query ITTT
 EXPLAIN SELECT a, b FROM t ORDER BY a, b

